app/models/product_test.rb

```diff
@@ -240,10 +253,10 @@ def pick_denom_ids
       end
       numer_ids.uniq!
       numer_ids.keep_if { |id| ApplicationController.helpers.mpl_id?(id) }
-      numer_ids = numer_ids.sample(50)
-      denom_ids = numer_ids + denom_ids.sample(50 - numer_ids.count)
+      numer_ids = numer_ids.sample(product.test_deck_max - 1)
+      denom_ids = numer_ids + denom_ids.sample(product.test_deck_max - numer_ids.count - 1)
     end
-    denom_ids
+    denom_ids << numer_id
```

Comment:
numer_id and numer_ids are a bit ambiguously named since they serve pretty different purposes. Can we rename one of the variables, or move the comment block to where numer_id is first defined and see if that makes it less ambiguous?